activerecord/test/cases/migration_test.rb

```diff
@@ -1133,10 +1133,12 @@ def test_adding_multiple_columns
           t.integer :age, default: 0
           t.date :birthdate, comment: "This is a comment"
           t.timestamps null: true
+          t.column :parent_thing, :bigint
+          t.foreign_key t.name, column: :parent_thing
```

Comment:
> otherwise this implementation also passes the tests

Wait, no it doesn't, I was running the tests against the SQLite adapter which skips this one 🤦 

This exposes another problem though: we can't return nil from a `_for_alter` method since it results in invalid SQL - but maybe we _should_ be able to do that.

Reply:
I split it into a separate test, that is also more specific with its asserts.

maybe we should handle nil returns for `_for_alter` methods, but for now I'm not looking into it. I verified that the two adapters that support bulk alter table (mysql and postgres) support foreign key add and drops in said bulk alters.